### PR TITLE
story/2033/3779/edit status menu

### PR DESCRIPTION
**Pull request by Xaviju**
Rama back: `us/2033/task/3788/back_endpoint_update_status`

### CODE REVIEW

**Comment by juanfran** [comment]
If I press "ESC" the change is saved.

**Comment by juanfran** [comment]
It's weird that if I click inside the tg-edit-status the form closes.

**Comment by juanfran** [comment]
The form horizontal alignment doesn't match the design, it should be in the same line as an unedited status.

**Comment by Xaviju** [comment]
Review fixed! :checkered_flag:

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      slug: '',
+    });
+
+    if (this.status) {
+      this.statusForm.get('name')?.setValue(this.status.name);
+      this.statusForm.get('slug')?.setValue(this.status.slug);
+    }
+
+    requestAnimationFrame(() => {
+      (this.statusInput?.nativeElement as HTMLElement)?.focus();
```
[the remark this answers is not in the document]
Comment (Xaviju) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts:122]: No, it does not focus on the element when the form appears!

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  public onActiveZoneChange(active: boolean) {
+    if (!active) {
+      this.leaveForm();
+    }
+  }
+
+  public editStatus() {
+    this.submitted = true;
+    this.cd.detectChanges();
```
[the remark this answers is not in the document]
Comment (Xaviju) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts:128]: This seems an outdated part of the refactor. Good catch!

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { NO_ERRORS_SCHEMA, signal } from '@angular/core';
+import { provideMockStore } from '@ngrx/store/testing';
+import { StatusMockFactory, WorkflowMockFactory } from '@taiga/data';
+import { getTranslocoModule } from '~/app/transloco/transloco-testing.module';
+import { EditStatusComponent } from './edit-status.component';
+
+const status = StatusMockFactory();
+const initialState = {};
+
+describe.only('EditStatusComponent', () => {
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.spec.ts:20]: only?

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanfran) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/edit-status/edit-status.component.ts:122]: I did a small test, and it seems to work for me

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/status/kanban-status.component.html [hunk 4]
```diff
@@ -21,15 +21,21 @@
...
         <tui-hosted-dropdown
-          *ngIf="vm.canEdit"
           class="status-options-menu"
           [tuiDropdownAlign]="'right'"
           [content]="projectActionsDropdown"
           [(open)]="projectActionsDropdownState">
           <button
-            [attr.aria-label]=""
+            *ngIf="vm.project.userIsAdmin"
+            [tgRestoreFocusTarget]="'leave-status-edition-' + status?.id"
```
Comment (Xaviju) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/status/kanban-status.component.html:36]: Fixed, the problem was that a `-` was missing here

**Review by juanfran — APPROVED** [review]

**Pull request merged**